packages/components/src/Select/Select.tsx

```diff
@@ -0,0 +1,259 @@
+import React, { useId, useState } from "react"
+import classnames from "classnames"
+import ReactSelect, {
+  components,
+  Props as ReactSelectProps,
+  NoticeProps,
+} from "react-select"
+import Async, { AsyncProps as ReactAsyncSelectProps } from "react-select/async"
+import { FieldMessage } from "~components/FieldMessage"
+import { ChevronDownIcon, ChevronUpIcon, ClearIcon } from "~components/Icon"
+import { Label } from "~components/Label"
+import { Tag } from "~components/Tag"
+import styles from "./Select.module.scss"
+
+export type SelectProps = {
+  /**
+   * The secondary variant is a more subdued variant that takes up as little space as possible
+   * `variant="secondary" reversed="false" is not implemented and will throw a "not implemented" error
+   * @default "default"
+   */
+  variant?: VariantType
+  status?: StatusType
+  label?: React.ReactNode
+  validationMessage?: React.ReactNode
+  description?: React.ReactNode
+  /**
+   * Use a reversed colour scheme
+   * @default false
+   */
+  reversed?: boolean
+  /**
+   * Whether the "select control" (the button you click to open the menu) width fills the
+   * full width of  the container or is as wide as the selected option text.
+   * Note that the control text will ellipsize if it is wider than the parent container.
+   * @default false
+   */
+  fullWidth?: boolean
+} & ReactSelectProps<any, boolean>
+
+export type VariantType = "default" | "secondary" | "secondary-small"
+
+export type StatusType = "default" | "error"
```

Comment:
Ah, missed these.

```suggestion
export type SelectVariant = "default" | "secondary" | "secondary-small"

export type SelectStatus = "default" | "error"
```

Alternatively (if we don't use them elsewhere) you can probably just put the strings directly into the prop